### PR TITLE
Fix the issue caused by the changed tensor names after complex transformation

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - Update logic of port name checking to check if queried port tensor names is subset instead of  op "==". 
 - *...*

### Tickets:
 - CVS-122932

### CODE REVIEW

File: src/inference/src/dev/isync_infer_request.cpp [hunk 1]
```diff
@@ -143,8 +143,7 @@ ov::ISyncInferRequest::FoundPort ov::ISyncInferRequest::find_port(const ov::Outp
     ov::ISyncInferRequest::FoundPort::Type type = ov::ISyncInferRequest::FoundPort::Type::INPUT;
     for (const auto& ports : {get_inputs(), get_outputs()}) {
         for (size_t i = 0; i < ports.size(); i++) {
-            if (ports[i].get_index() == port.get_index() && ports[i].get_names() == port.get_names() &&
-                check_nodes(ports[i].get_node(), port.get_node())) {
+            if (ports[i].get_index() == port.get_index() && check_nodes(ports[i].get_node(), port.get_node())) {
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/isync_infer_request.cpp:146]: do we have a list of transformations where i/o tensor names are changed?

**Comment by ilya-lavrenov** [comment]
If you could eliminate usage of tensor names here https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/infer_request.cpp#L30, I think it will help. 

It's related to CVS-132085

File: src/inference/src/dev/isync_infer_request.cpp (hunk 1 above)
Comment (yangwang201911) [reply src/inference/src/dev/isync_infer_request.cpp:146]: Not yet. For this model "text_encoder" from Stable Diffusion pipeline in my reproduction, the transformation, [`EliminateReshape`](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp#L331), changed the input tensor names.

Comment (ilya-lavrenov) [reply src/inference/src/dev/isync_infer_request.cpp:146]: could you please create a bug for this transformation?

Comment (yangwang201911) [reply src/inference/src/dev/isync_infer_request.cpp:146]: Sure.

Comment (yangwang201911) [reply src/inference/src/dev/isync_infer_request.cpp:146]: 133925

**Comment by yangwang201911** [comment]
> If you could eliminate usage of tensor names here https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/infer_request.cpp#L30, I think it will help.
> 
> It's related to [CVS-132085](https://jira.devtools.intel.com/browse/CVS-132085)

Will try this soon.

**Comment by yangwang201911** [comment]
> > If you could eliminate usage of tensor names here https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/infer_request.cpp#L30, I think it will help.
> > It's related to [CVS-132085](https://jira.devtools.intel.com/browse/CVS-132085)
> 
> Will try this soon.

has synced with @zhaixuejun1993 and @riverlijunjie and [PR23035 ](https://github.com/openvinotoolkit/openvino/pull/23035)will remove the usage of tensor / operation names within CVS-132085

File: src/inference/src/dev/isync_infer_request.cpp (hunk 1 above)
Comment (songbell) [reply src/inference/src/dev/isync_infer_request.cpp:146]: maybe for now to fix the issue in jira, we can change the port name "==" to logic like this: the compiled model port names should include the query port names, I suppose this is the minimum impact to existing code logic.

Comment (ilya-lavrenov) [reply src/inference/src/dev/isync_infer_request.cpp:146]: > we can change the port name "==" to logic like this: the compiled model port names should include the query port names

Agree with it as more concervative solution

Comment (yangwang201911) [reply src/inference/src/dev/isync_infer_request.cpp]: Updated.

File: src/inference/tests/unit/iplugin_test.cpp [hunk 2]
```diff
@@ -81,6 +81,42 @@ MATCHER_P(blob_in_map_pointer_is_same, ref_blob, "") {
...
+    EXPECT_THROW(mock_infer_request->set_tensor(updated_param->output(0), tensor), ov::Exception);
+
+    updated_param->output(0).set_names({"param", "new_name"});
+    EXPECT_THROW(mock_infer_request->set_tensor(updated_param->output(0), tensor), ov::Exception);
+
+    updated_param->output(0).set_names({"new_name", "name2"});
+    EXPECT_THROW(mock_infer_request->set_tensor(updated_param->output(0), tensor), ov::Exception);
+}
+
+TEST_F(IPluginTest, SetCorrectTensorNameForPort) {
```
Comment (songbell) [inline src/inference/tests/unit/iplugin_test.cpp:99]:
```suggestion
TEST_F(IPluginTest, SetTensorWithCorrectPortNames) {
```

File: src/inference/tests/unit/iplugin_test.cpp [hunk 3]
```diff
@@ -81,6 +81,42 @@ MATCHER_P(blob_in_map_pointer_is_same, ref_blob, "") {
     return reinterpret_cast<float*>(arg.begin()->second->buffer()) == reinterpret_cast<float*>(ref_blob->buffer());
 }
 
+TEST_F(IPluginTest, SetIncorrectTensorNameForPort) {
```
Comment (songbell) [inline src/inference/tests/unit/iplugin_test.cpp:84]:
```suggestion
TEST_F(IPluginTest, failToSetTensorWithIncorrectPortNames) {
```

**Review by songbell — APPROVED** [review]

File: src/inference/tests/unit/iplugin_test.cpp (hunk 3 above)
Comment (yangwang201911) [reply src/inference/tests/unit/iplugin_test.cpp]: Updated

File: src/inference/src/dev/isync_infer_request.cpp [hunk 4]
```diff
@@ -121,6 +121,16 @@ const std::shared_ptr<const ov::ICompiledModel>& ov::ISyncInferRequest::get_comp
 }
 
 ov::ISyncInferRequest::FoundPort ov::ISyncInferRequest::find_port(const ov::Output<const ov::Node>& port) const {
+    // check if the tensor names of target port is a subset of source port's tensor names
+    auto check_tensor_names = [](const std::unordered_set<std::__cxx11::string>& source,
+                                 const std::unordered_set<std::__cxx11::string>& target) {
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/isync_infer_request.cpp]: why do we need `__cxx11` ?

Comment (yangwang201911) [reply src/inference/src/dev/isync_infer_request.cpp]: have updated.

**Pull request merged**